agent/handlers/v4/container_metadata_handler.go

```diff
@@ -89,25 +89,49 @@ func GetContainerNetworkMetadata(containerID string, state dockerstate.TaskEngin
 	if settings == nil {
 		return nil, errors.Errorf("unable to generate network response for container '%s'", containerID)
 	}
-	// This metadata is the information provided in older versions of the API
-	// We get the NetworkMode (Network interface name) from the HostConfig because this
-	// this is the network with which the container is created
-	ipv4AddressFromSettings := settings.IPAddress
-	networkModeFromHostConfig := dockerContainer.Container.GetNetworkMode()
 
 	// Extensive Network information is not available for Docker API versions 1.17-1.20
 	// Instead we only get the details of the first network
 	networks := make([]Network, 0)
 	if len(settings.Networks) > 0 {
 		for modeFromSettings, containerNetwork := range settings.Networks {
 			networkMode := modeFromSettings
-			ipv4Addresses := []string{containerNetwork.IPAddress}
-			network := Network{Network: containermetadata.Network{NetworkMode: networkMode, IPv4Addresses: ipv4Addresses}}
+			ipv4Addresses := []string{
```

Comment:
i don't think you need to change anything in GetContainerNetworkMetadata, since its only used by non-awsvpc task https://github.com/aws/amazon-ecs-agent/blob/master/agent/handlers/v4/container_metadata_handler.go#L71. for awsvpc task, the ipv6 address is already in v2 container response i think https://github.com/aws/amazon-ecs-agent/blob/master/agent/handlers/v2/response.go#L242

Reply:
updated.